arches_component_lab/src/arches_component_lab/widgets/ResourceInstanceSelectWidget/components/ResourceInstanceSelectWidgetEditor.vue

```diff
@@ -0,0 +1,223 @@
+
+const props = defineProps<{
+    initialValue: ResourceInstanceReference | null | undefined;
+    graphSlug: string;
+    nodeAlias: string;
+}>();
+
+const { $gettext } = useGettext();
+
+const itemSize = 36; // in future iteration this should be declared in the CardXNodeXWidget config
+
+const options = ref<ResourceInstanceReference[]>([]);
+const isLoading = ref(false);
+const resourceResultsPage = ref(0);
+const resourceResultsTotalCount = ref(0);
+const fetchError = ref<string | null>(null);
+
+const resourceResultsCurrentCount = computed(() => options.value.length);
+
+onMounted(async () => {
+    await getOptions(1);
+});
+
+function clearOptions() {
+    options.value = props.initialValue ? [props.initialValue] : [];
+}
+
+function onFilter(event: MultiSelectFilterEvent) {
+    clearOptions();
+    getOptions(1, event.value);
+}
+
+async function getOptions(page: number, filterTerm?: string) {
+    try {
+        isLoading.value = true;
+
+        const resourceData = await fetchRelatableResources(
+            props.graphSlug,
+            props.nodeAlias,
+            page,
+            filterTerm,
+            props.initialValue ? [props.initialValue] : null,
+        );
+
+        const references = resourceData.data.map(
+            (
+                resourceRecord: ResourceInstanceResult,
+            ): ResourceInstanceReference => ({
+                display_value: resourceRecord.display_value,
+                resourceId: resourceRecord.resourceinstanceid,
+                ontologyProperty: "",
+                inverseOntologyProperty: "",
+            }),
+        );
+
+        if (resourceData.current_page == 1) {
+            options.value = references;
+        } else {
+            options.value = [...options.value, ...references];
+        }
+
+        resourceResultsPage.value = resourceData.current_page;
+        resourceResultsTotalCount.value = resourceData.total_results;
+    } catch (error) {
+        fetchError.value = (error as Error).message;
+    } finally {
+        isLoading.value = false;
+    }
+}
+
+async function onLazyLoadResources(event?: VirtualScrollerLazyEvent) {
+    if (isLoading.value) {
+        return;
+    }
+
+    if (
+        // if we have already fetched all the resources
+        resourceResultsTotalCount.value > 0 &&
+        resourceResultsCurrentCount.value >= resourceResultsTotalCount.value
+    ) {
+        return;
+    }
+
+    if (
+        // if the user has NOT scrolled to the end of the list
+        event &&
+        event.last < resourceResultsCurrentCount.value - 1
+    ) {
+        return;
+    }
+
+    if (
+        // if the dropdown is opened and we already have data
+        !event &&
+        resourceResultsCurrentCount.value > 0
+    ) {
+        return;
+    }
+
+    await getOptions((resourceResultsPage.value || 0) + 1);
+}
+
+function resolver(e: FormFieldResolverOptions) {
+    validate(e);
+
+    let value = e.value;
+
+    if (!Array.isArray(value)) {
```

Comment:
while multi-select coerces data into an array, since this is single-select I think the data coercion should match the expected shape: a single resource.